src/libasr/codegen/x86_assembler.cpp

```diff
@@ -256,12 +256,14 @@ void emit_elf64_header(X86Assembler &a, uint32_t p_flags) {
     a.asm_dq_imm64(0x1000);   // p_align
 
     a.add_var("phdrsize", a.pos()-a.get_defined_symbol("phdr").value);
-    a.add_var("e_phoff", a.get_defined_symbol("phdr").value-a.origin());
+    a.add_var64("e_phoff", a.get_defined_symbol("phdr").value-a.origin());
 }
 
 void emit_elf64_footer(X86Assembler &a) {
-    a.add_var("e_entry", a.get_defined_symbol("_start").value);
-    a.add_var("filesize", a.pos()-a.origin());
+    a.define_symbol("filesize", a.pos()-a.origin());
+    a.append_asm("\n\nfilesize equ $ - $$"); // $ refers current address, $$ refers starting address
```

Comment:
We can create a function:
```
a.add_var_size("filesize")
```
Which would emit `filesize equ $ - $$` in string assembly, but an equivalent of `a.add_var("filesize", a.pos()-a.origin());`.

Reply:
Here is how it can be implemented:
```c++
#    define EMIT_VAR_SIZE(a) emit("\n", a + " equ $ - $$\n")

    void add_var_size(const std::string &var) {
        uint64_t val = pos() - origin();
        define_symbol(var, val);
        EMIT_VAR_SIZE(var);
    }
```

Reply:
Got it. Thank you for the guidance. I added this `add_var_size()` function.